geode-core/src/main/java/org/apache/geode/internal/security/IntegratedSecurityService.java

```diff
@@ -119,8 +121,12 @@ public Subject getSubject() {
       }
     }
 
-    // in other cases like rest call, client operations, we get it from the current thread
-    currentUser = SecurityUtils.getSubject();
+    try {
+      // in other cases like rest call, client operations, we get it from the current thread
+      currentUser = SecurityUtils.getSubject();
+    } catch (UnavailableSecurityManagerException e) {
+      throw new CacheClosedException("likely cache shutdown in progress", e);
```

Comment:
Is it possible to check the value of `GemfireCahceImpl.isClosing` here to confirm if we are actually closing the cache, rather than just assuming that we are?

Reply:
At this point a `cache` object is not in scope and given that the issue we're addressing here is related to race conditions with cache requests on the one hand, and cache being either gone or in the process of being shut down on the other, I don't know if it's worth adding the functionality of bringing it in scope and checking its state.